packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/scroll/ReactHorizontalScrollContainerLegacyView.kt

```diff
@@ -19,17 +19,17 @@ internal class ReactHorizontalScrollContainerLegacyView(context: Context) :
     ReactViewGroup(context) {
   private val isRTL: Boolean = I18nUtil.instance.isRTL(context)
 
-  override fun setRemoveClippedSubviews(removeClippedSubviews: Boolean) {
-    // removeClippedSubviews logic may read metrics before the offsetting we do in onLayout() and is
-    // such unsafe
-    if (isRTL) {
-      super.setRemoveClippedSubviews(false)
-      return
+  override var removeClippedSubviews: Boolean = false
+    set(value) {
+      // removeClippedSubviews logic may read metrics before the offsetting we do in onLayout() and
+      // is such unsafe
+      if (isRTL) {
+        field = false
+      } else {
+        field = value
+      }
```

Comment:
I think there are issue with this code.
1) `override var removeClippedSubviews: Boolean = false` created a new backing field for `ReactHorizontalScrollContainerLegacyView.kt` instead of using parent class.
So remove the `= false` and add `get()` as it will give error without it.

2) You need to use `super.removeClippedSubviews` to access parent field but if you use `field` it is setting the value on the backing field for the current class and not the parent.

```suggestion
  override var removeClippedSubviews: Boolean
    set(value) {
      // removeClippedSubviews logic may read metrics before the offsetting we do in onLayout() and
      // is such unsafe
      if (isRTL) {
        super. removeClippedSubviews = false
      } else {
        super. removeClippedSubviews = value
      }
```

I'll test this out and report result later.

Reply:
Thank you so much for the thorough explanation and for checking this in-depth; it makes a lot of sense now. Sorry for the back-and-forth on this one :)

I've applied these changes in 6fd5421b12ee95cefe94fca273c35c7f71cd35a4.